src/SapAct/Workers/SapActBaseWorker.cs

```diff
@@ -92,19 +92,31 @@ private async Task ProcessMessageAsync(ServiceBusReceivedMessage message, Cancel
         if (message == null || message.Body == null)
             return;
 
-        JsonDocument jsonDocument = JsonDocument.Parse(Encoding.UTF8.GetString(message.Body));
+		JsonDocument jsonDocument = JsonDocument.Parse(Encoding.UTF8.GetString(message.Body));
 
-		for (int x = 0; x < jsonDocument.RootElement.GetArrayLength(); x++) //TODO: this is temporary, array not expected
+		if (jsonDocument.RootElement.ValueKind == JsonValueKind.Array)
```

Comment:
Could potentially simplify the flow here by lifting the object case into the array case?

e.g.
```
var items = jsonDocument.RootElement.ValueKind switch
{
  JsonValueKind.Array => jsonDocument.RootElement.EnumerateArray(),
  JsonValueKind.Object => [jsonDocument.RootElement],
  _ => throw new ApplicationException("Unexpected message format")
};

foreach (var item in items)
{
  await IngestMessageAsync(message, cancellationToken);
  telemetryClient.TrackMetric(GetTelemetryMetric(message.MessageId));
}
```

Reply:
yeah I like this - economical